neonvm/controllers/metrics.go

```diff
@@ -68,15 +68,46 @@ type wrappedReconciler struct {
 	failing map[client.ObjectKey]struct{}
 }
 
+// ReocncileSnapshot provides a glimpse into the current state of ongoing reconciles
+//
+// This type is (transitively) returned by the controller's "dump state" HTTP endpoint, and exists
+// to allow us to get deeper information on the metrics - we can't expose information for every
+// VirtualMachine into the metrics (it'd be too high cardinality), but we *can* make it available
+// when requested.
+type ReconcileSnapshot struct {
+	// ControllerName is the name of the controller: virtualmachine or virtualmachinemigration.
+	ControllerName string
+
+	// Failing is the list of objects currently failing to reconcile
+	Failing []string
+}
+
 // WithMetrics wraps a given Reconciler with metrics capabilities.
```

Comment:
Good to update the comment here mentioning we return a snapshot function as well.

Reply:
done, in bee7db9e3097aa0588d6d6d4203e25d7adbd2bb6